NO-JIRA: Improve must-gather image handling and annotation checks- #2071

midu16 opened the pull request:
## Pull Request: Improve must-gather image handling and annotation checks

###  Summary

This PR enhances the `MustGatherOptions.completeImages()` method by:

* **Logging warnings** for CSVs that are missing the required `must-gather-image` annotation.
* Improving robustness and readability of the image collection logic.
* Ensuring deduplication of default must-gather image when collecting plug-in images.


### Changes

* Moved CSV annotation scanning inline to allow per-CSV warning messages.
* Added warning log in the format:

  ```
  WARNING: CSV operator <name> doesn't have the must-gather-image annotation.
  ```
* Enhanced image resolution logic to:

  * Fallback to hardcoded must-gather image if the image stream tag cannot be resolved.
  * Log warnings for such fallbacks.
* Maintained support for `--all` (aka `AllImages`) by collecting images from both:

  * CSVs (ClusterServiceVersions)
  * ClusterOperators
* Ensured duplicate images (especially the default one) are not added to the final list.


### Testing

* Verified that missing annotations on CSVs produce proper warning output.
* Confirmed image collection and deduplication works as expected.
* Manually validated fallback logic for missing default image stream tag.


### Related

* Annotation key used: `operators.openshift.io/must-gather-image`
* Fulfills a need for better diagnostics and operator plug-in visibility in `oc adm must-gather --all-images`.

---

### Notes

* Future enhancement may consider moving CSV image collection into a separate method again, but returning both:

  * `pluginImages`
  * `[]string` of CSVs missing the annotation for external logging.

midu16 commented:
> Hi @midu16 , can you please check the CI job failures. `# github.com/openshift/oc/pkg/cli/admin/mustgather` `pkg/cli/admin/mustgather/mustgather.go:290:23: undefined: csvs`

@shivprakashmuley can you review now ?

midu16 commented:
hi @ardaguclu i wonder if you can help me with this PR ?

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -272,6 +272,7 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 			return fmt.Errorf("unable to resolve image stream '%v': %v", imageStream, err)
 		}
 	}
+	// If no images or --all flag is set, use default must-gather image
```
Redundant change?

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -288,12 +289,12 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 		pluginImages, err = o.annotatedCSVs(ctx)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to list CSVs: %v", err)
```
```suggestion
			return fmt.Errorf("failed to list CSVs: %w", err)
```

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -288,12 +289,12 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 		cos, err := o.ConfigClient.ConfigV1().ClusterOperators().List(ctx, metav1.ListOptions{})
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to list ClusterOperators: %v", err)
```
We don't need this. We are already wrapping the error above.

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -329,6 +330,8 @@ func (o *MustGatherOptions) annotatedCSVs(ctx context.Context) (map[string]struc
 		if v, ok := ann[mgAnnotation]; ok {
 			pluginImages[v] = struct{}{}
+		} else {
+			o.log("WARNING: CSV operator %s doesn't have the must-gather-image annotation.", item.GetName())
```
Logging this per image would flood the logs. We can collect the ones that do not have this annotation and log once.

ardaguclu reviewed (commented):
Dropped a few comments. Can you please test this on real OCP cluster and paste the log here to see how many operators lack the annotation, so that we can see what customers see in real cluster.

midu16 committed:
Fix undefined csvs in mustgather and add CSV annotation warning

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -342,12 +342,18 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 			}
 		}
+		if len(missingAnnotation) > 0 {
+			o.log("WARNING: ClusterOperators without %s annotation: %s", mgAnnotation, strings.Join(missingAnnotation, ", "))
```
This is redundant. That would print duplicated messages. It is better to only have the log in annotated function

ardaguclu replied:
This is more informative rather than being warning. So can you please drop the WARNING prefix.

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -372,12 +378,18 @@ func (o *MustGatherOptions) annotatedCSVs(ctx context.Context) (map[string]struc
 		}
 	}
+	if len(missingAnnotation) > 0 {
+		o.log("WARNING: CSVs without %s annotation: %s", mgAnnotation, strings.Join(missingAnnotation, ", "))
```
Logging can be managed in upper level near to ClusterOperators. So you can return the images that missing the annotation. After that;

```go
 if len(missingCOs) > 0 {
     o.log("ClusterOperators without %s annotation: %s", mgAnnotation, strings.Join(missingCOs, ", "))
  }
  if len(missingCSVs) > 0 {
      o.log("CSVs without %s annotation: %s", mgAnnotation, strings.Join(missingCSVs, ", "))
  }

```

Commit:
must-gather: move missing-annotation logging to caller

Return missing annotation names from annotatedCSVs instead of logging
internally, so both ClusterOperator and CSV messages are logged at the
same level in completeImages. Drop the WARNING prefix since the messages
are informational.

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -341,7 +341,7 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 	if o.AllImages {
 		// find all csvs and clusteroperators with the annotation "operators.openshift.io/must-gather-image"
-		pluginImages, err := o.annotatedCSVs(ctx)
+		pluginImages, missingCSVs, err := o.annotatedCSVs(ctx)
```
```suggestion
		pluginImages, annotationMissingCSVs, err := o.annotatedCSVs(ctx)
```

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -350,12 +350,21 @@ func (o *MustGatherOptions) completeImages(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		var missingCOs []string
```
```suggestion
		var annotationMissingCOs []string
```

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -377,16 +386,19 @@ func (o *MustGatherOptions) annotatedCSVs(ctx context.Context) (map[string]struc
+		return nil, nil, err
 	}
 
+	var missingAnnotation []string
```
```suggestion
	var annotationMissingImages []string
```

Commit:
must-gather: improve missing-annotation variable naming

Rename missingCOs/missingCSVs to annotationMissingCOs/annotationMissingCSVs
to clarify these track resources missing the must-gather annotation.

Commit:
must-gather: rename missingAnnotation to annotationMissingImages

Align internal variable naming in annotatedCSVs with the caller-side
convention (annotationMissingCSVs, annotationMissingCOs) per review
feedback.

ardaguclu commented:
/label tide/merge-method-squash
/lgtm
/approve

ardaguclu commented:
/retitle NO-JIRA: Improve must-gather image handling and annotation checks- #2071

ardaguclu commented:
/retitle NO-JIRA: Improve must-gather image handling and annotation checks

midu16 commented:
/verified by @midu16

[test_must_gather_annotation_warning.sh](https://github.com/user-attachments/files/30580805/test_must_gather_annotation_warning.sh)


# Must-Gather Annotation Warning — Test Results

**Date:** 2026-07-31  
**Binary:** `./oc` (v4.2.0-alpha.0-3022-gafbcfb7, commit afbcfb7cf)  
**Cluster:** OpenShift 4.18.27 (`KUBECONFIG=./kubeconfig`, user `system:admin`)

## Changes Under Test

Commits `cf8b4e273..afbcfb7cf` modified `oc adm must-gather --all-images` to log informational warnings when ClusterOperators or CSVs lack the `operators.openshift.io/must-gather-image` annotation. Previously these operators were silently skipped.

| Commit | Description |
|--------|-------------|
| cf8b4e2 | Fix undefined csvs in mustgather and add CSV annotation warning |
| 2c804f3 | Move missing-annotation logging to caller |
| 4b4339a | Improve missing-annotation variable naming |
| afbcfb7 | Rename missingAnnotation to annotationMissingImages |

## Cluster State

| Resource | With Annotation | Without Annotation |
|----------|-----------------|--------------------|
| ClusterOperators | 0 | 34 |
| CSVs (unique names) | 5 | 12 |

### Annotated CSV Images

| CSV | Image |
|-----|-------|
| advanced-cluster-management.v2.13.10 | `registry.redhat.io/rhacm2/acm-must-gather-rhel9@sha256:f12f01...` |
| oadp-operator.v1.4.10 | `registry.redhat.io/oadp/oadp-mustgather-rhel9@sha256:870009...` |
| local-storage-operator.v4.18.0-202607171826 | `registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:ea72be...` |
| cluster-logging.v6.2.12 | `registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:534f27...` |
| ocs-operator.v4.18.25-rhodf | `registry.redhat.io/odf4/odf-must-gather-rhel9@sha256:51e0bb...` |

## Test Results

| # | Test | Result |
|---|------|--------|
| 1 | ClusterOperator annotation warning line is present | PASS |
| 2 | All 34 unannotated ClusterOperators listed in warning | PASS |
| 3 | CSV annotation warning line is present | PASS |
| 4 | All 12 unique unannotated CSV names appear in warning | PASS |
| 5 | `Using must-gather plug-in image` line is present | PASS |
| 6 | All 5 annotated CSV images included in plug-in list | PASS |
| 7 | ClusterOperator warning appears before image list (line 1 < 3) | PASS |
| 8 | CSV warning appears before image list (line 2 < 3) | PASS |

**8 passed, 0 failed**

## Log Output: Before vs After

### Before (commit a4c90c71, no annotation warnings)

No indication of which operators lack the must-gather-image annotation. The output jumps
straight from image resolution to the cluster summary and pod creation:

```
[must-gather      ] OUT 2026-07-31T08:35:39.025348396Z Using must-gather plug-in image:
    quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a2c8408d...,
    registry.redhat.io/oadp/oadp-mustgather-rhel9@sha256:87000900...,
    registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:ea72befb...,
    registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:534f2752...,
    registry.redhat.io/odf4/odf-must-gather-rhel9@sha256:51e0bbde...,
    registry.redhat.io/rhacm2/acm-must-gather-rhel9@sha256:f12f016d...
When opening a support case, bugzilla, or issue please include the following summary data
along with any other requested information:
ClusterID: 6bb33e72-07a1-4284-ab9f-05516d469689
ClientVersion: unknown
ClusterVersion: Stable at "4.18.27"
ClusterOperators:
	All healthy and stable


[must-gather      ] OUT 2026-07-31T08:35:39.482811773Z namespace/openshift-must-gather-nc4wq created
[must-gather      ] OUT 2026-07-31T08:35:39.614929398Z clusterrolebinding.rbac.authorization.k8s.io/must-gather-sd7bk created
[must-gather      ] OUT 2026-07-31T08:35:39.907347642Z pod for plug-in image
    registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:534f2752... created
[must-gather      ] OUT 2026-07-31T08:35:39.90738414Z pod for plug-in image
    registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:ea72befb... created
[must-gather      ] OUT 2026-07-31T08:35:39.911311106Z pod for plug-in image
    registry.redhat.io/oadp/oadp-mustgather-rhel9@sha256:87000900... created
[must-gather      ] OUT 2026-07-31T08:35:39.911372451Z pod for plug-in image
    quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a2c8408d... created
```

### After (commit afbcfb7, with annotation warnings)

Two new informational lines appear **before** the `Using must-gather plug-in image` line,
listing every ClusterOperator and CSV that lacks the `operators.openshift.io/must-gather-image`
annotation:

```
[must-gather      ] OUT 2026-07-31T08:34:48.344870563Z ClusterOperators without
    operators.openshift.io/must-gather-image annotation: authentication, baremetal,
    cloud-controller-manager, cloud-credential, cluster-autoscaler, config-operator, console,
    control-plane-machine-set, csi-snapshot-controller, dns, etcd, image-registry, ingress,
    insights, kube-apiserver, kube-controller-manager, kube-scheduler,
    kube-storage-version-migrator, machine-api, machine-approver, machine-config, marketplace,
    monitoring, network, node-tuning, olm, openshift-apiserver, openshift-controller-manager,
    openshift-samples, operator-lifecycle-manager, operator-lifecycle-manager-catalog,
    operator-lifecycle-manager-packageserver, service-ca, storage
[must-gather      ] OUT 2026-07-31T08:34:48.344962054Z CSVs without
    operators.openshift.io/must-gather-image annotation: openshift-gitops-operator.v1.15.4,
    topology-aware-lifecycle-manager.v4.18.5, packageserver,
    cephcsi-operator.v4.18.25-rhodf, mcg-operator.v4.18.25-rhodf,
    ocs-client-operator.v4.18.25-rhodf, odf-csi-addons-operator.v4.18.25-rhodf,
    odf-dependencies.v4.18.25-rhodf, odf-operator.v4.18.25-rhodf,
    odf-prometheus-operator.v4.18.25-rhodf, recipe.v4.18.25-rhodf,
    rook-ceph-operator.v4.18.25-rhodf, ...
[must-gather      ] OUT 2026-07-31T08:34:48.344978555Z Using must-gather plug-in image:
    quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a2c8408d...,
    registry.redhat.io/oadp/oadp-mustgather-rhel9@sha256:87000900...,
    registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:ea72befb...,
    registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:534f2752...,
    registry.redhat.io/odf4/odf-must-gather-rhel9@sha256:51e0bbde...,
    registry.redhat.io/rhacm2/acm-must-gather-rhel9@sha256:f12f016d...
When opening a support case, bugzilla, or issue please include the following summary data
along with any other requested information:
ClusterID: 6bb33e72-07a1-4284-ab9f-05516d469689
ClientVersion: unknown
ClusterVersion: Stable at "4.18.27"
ClusterOperators:
	All healthy and stable


[must-gather      ] OUT 2026-07-31T08:34:48.811666519Z namespace/openshift-must-gather-g74xh created
[must-gather      ] OUT 2026-07-31T08:34:48.959852123Z clusterrolebinding.rbac.authorization.k8s.io/must-gather-qkk4f created
[must-gather      ] OUT 2026-07-31T08:34:49.259808718Z pod for plug-in image
    registry.redhat.io/oadp/oadp-mustgather-rhel9@sha256:87000900... created
[must-gather      ] OUT 2026-07-31T08:34:49.268102002Z pod for plug-in image
    registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:534f2752... created
[must-gather      ] OUT 2026-07-31T08:34:49.268135525Z pod for plug-in image
    quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a2c8408d... created
[must-gather      ] OUT 2026-07-31T08:34:49.268271219Z pod for plug-in image
    registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:ea72befb... created
```

### Key Differences

| Aspect | Before | After |
|--------|--------|-------|
| ClusterOperator missing-annotation warning | Not emitted | Emitted before image list |
| CSV missing-annotation warning | Not emitted | Emitted before image list |
| `Using must-gather plug-in image` line | First output line | Third output line |
| Logging format | `[must-gather] OUT <timestamp> <message>` | Same format, consistent |
| Image selection | Unchanged | Unchanged |
| Pod creation | Unchanged | Unchanged |

## How to Reproduce

```bash
make oc   # or: go build -mod=vendor -tags 'include_gcs include_oss containers_image_openpgp' -o ./oc ./cmd/oc/
export KUBECONFIG=./kubeconfig
bash ./test_must_gather_annotation_warning.sh
```

ardaguclu commented:
/hold

ardaguclu commented:
/lgtm cancel
we first need to ensure that we don't flood the logs

ardaguclu commented:
/approve cancel

Commit:
must-gather: deduplicate CSV warning and remove CO warning

ClusterOperators are platform-level components that never carry the
must-gather-image annotation—listing all 34 of them on every
--all-images run is noise. Remove the CO warning entirely and keep
the annotation check only for day-2 operators (CSVs).

CSVs exist in every namespace, producing 100+ duplicated entries in
the warning line. Use a map[string]struct{} set to deduplicate names
and slices.Sorted(maps.Keys(...)) for deterministic, sorted output.

midu16 commented:
@ardaguclu the new change: 

`midu@midu-thinkpadp16vgen1:~/oc$ ./oc adm must-gather --all-images
[must-gather      ] OUT 2026-07-31T14:19:59.509617907Z CSVs without operators.openshift.io/must-gather-image annotation: cephcsi-operator.v4.21.9-rhodf, mcg-operator.v4.21.9-rhodf, oadp-operator.v1.6.1, ocs-client-operator.v4.21.9-rhodf, odf-csi-addons-operator.v4.21.9-rhodf, odf-dependencies.v4.21.9-rhodf, odf-external-snapshotter-operator.v4.21.9-rhodf, odf-operator.v4.21.9-rhodf, odf-prometheus-operator.v4.21.9-rhodf, packageserver, quay-operator.v3.16.5, recipe.v4.21.9-rhodf, rook-ceph-operator.v4.21.9-rhodf, topology-aware-lifecycle-manager.v4.22.0
[must-gather      ] OUT 2026-07-31T14:19:59.509637554Z Using must-gather plug-in image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:247045b6df3e9f7ccc796c6df492be3fa1df4937c284fbb5a08a6614af0f93e1, registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9@sha256:8123969ba984e8ca09649e144bcde15187107347d6562adde4a20cec6092944c, registry.redhat.io/odf4/odf-must-gather-rhel9@sha256:009aa228f54b26c3b4808d8609bb365eb73e050e56a2d713d46c4c8bb94bb45d, registry.redhat.io/openshift-gitops-1/must-gather-rhel9@sha256:36cfce9298bee2a7c81e8b120a6cc28f3e5e2f4656abc9cf6aa7a5e377b6249a, registry.redhat.io/multicluster-engine/must-gather-rhel9@sha256:e5db11243d9db5c27989c6da46ed473b6b7ff19ad903e9d759c20db82169b40c, registry.redhat.io/rhacm2/acm-must-gather-rhel9@sha256:1a4f21543e04c163a27ffba289dcab16464b014c20a625189c8e878c8534467d
When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
ClusterID: 69b6e1de-49e8-405c-ba5c-f8aeb5069d85
ClientVersion: unknown
ClusterVersion: Stable at "4.22.5"
ClusterOperators:
	clusteroperator/operator-lifecycle-manager is not upgradeable because ClusterServiceVersions blocking minor version upgrades to 4.23 or major version upgrades to 5.0:
- maximum supported OCP version for openshift-storage/odf-operator.v4.21.9-rhodf is 4.22
- maximum supported OCP version for openshift-storage/odf-dependencies.v4.21.9-rhodf is 4.22


[must-gather      ] OUT 2026-07-31T14:19:59.92175445Z namespace/openshift-must-gather-dmn8f created
[must-gather      ] OUT 2026-07-31T14:20:00.057629817Z clusterrolebinding.rbac.authorization.k8s.io/must-gather-twzrk created
[must-gather      ] OUT 2026-07-31T14:20:00.445839017Z pod for plug-in image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:247045b6df3e9f7ccc796c6df492be3fa1df4937c284fbb5a08a6614af0f93e1 created
`

ardaguclu commented on pkg/cli/admin/mustgather/mustgather.go:
```diff
@@ -377,16 +382,19 @@ func (o *MustGatherOptions) annotatedCSVs(ctx context.Context) (map[string]struc
+		return nil, nil, err
 	}
 
+	annotationMissingImages := make(map[string]struct{})
```
Set would be a better data type for this

Commit:
must-gather: use sets.Set for CSV deduplication

Replace map[string]struct{} with k8s.io/apimachinery/pkg/util/sets.Set
for deduplicating CSV names missing the must-gather-image annotation.
sets.List() returns a sorted slice, replacing slices.Sorted(maps.Keys()).

ardaguclu commented:
/hold cancel
can you please test this on 5.0 cluster and drop your results here?
/lgtm
/approve

midu16 commented:
@ardaguclu 

## Run 2: OpenShift 5.0.0-ec.5

**Date:** 2026-08-04
**Binary:** `./oc` (built from local working tree, commit ac0495e8a)
**Cluster:** OpenShift 5.0.0-ec.5 (SNO `zt-sno1.jade.bos2.lab`)
**User:** `system:admin`

### Cluster State

| Resource | With Annotation | Without Annotation |
|----------|-----------------|--------------------|
| ClusterOperators | 0 | 34+ (8 missing) |
| CSVs (unique names) | 3 | 2 |

#### CSVs WITHOUT annotation

| CSV |
|-----|
| packageserver |
| sriov-network-operator.v5.0.0-202607301230 |

#### CSVs WITH annotation (images)

| CSV | Image |
|-----|-------|
| lvms-operator | `registry.redhat.io/lvms4/lvms-must-gather-rhel9@sha256:92ba27...` |
| ptp-operator | `registry.redhat.io/openshift5/ptp-must-gather-rhel9@sha256:5a49c7...` |
| cluster-logging | `registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:1b097b...` |

### Test Results

| # | Test | Result |
|---|------|--------|
| 1 | No ClusterOperator warning emitted (correct - only day-2 operators matter) | PASS |
| 2 | CSV annotation warning line is present | PASS |
| 3 | All 2 unique unannotated CSV names appear in warning | PASS |
| 4 | CSV names are deduplicated (2 names, matches 2 unique) | PASS |
| 5 | CSV names are sorted alphabetically | PASS |
| 6 | `Using must-gather plug-in image` line is present | PASS |
| 7 | All 3 annotated CSV images included in plug-in list | PASS |
| 8 | CSV warning appears before image list (line 1 < 2) | PASS |

**8 passed, 0 failed**

### Must-Gather Output

```
[must-gather      ] OUT 2026-08-04T13:24:02.549590341Z CSVs without operators.openshift.io/must-gather-image annotation: packageserver, sriov-network-operator.v5.0.0-202607301230
[must-gather      ] OUT 2026-08-04T13:24:02.54962212Z Using must-gather plug-in image: quay.io/openshift-release-dev/ocp-v5.0-art-dev@sha256:be149b15f5a5658e85f1eee8a715bcff9c66e5fa5866f5f9e58e09208a63a673, registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:1b097b1a7630af0d78a2405d2cdd63282099c2e658a018f4b5080271d8c6f604, registry.redhat.io/openshift5/ptp-must-gather-rhel9@sha256:5a49c7d3587cac886a96a8a2f60413ab7963c633b039a314ba4433a6173b3706, registry.redhat.io/lvms4/lvms-must-gather-rhel9@sha256:92ba27b93c4bc0180b4b61cf3cfd40330755133b8abc45d888f2ee6b6f4b73b5
When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
ClusterID: c00dd50a-5771-40e2-8cb2-c009cd4d86fa
ClientVersion: unknown
ClusterVersion: Stable at "5.0.0-ec.5"
ClusterOperators:
	clusteroperator/config-operator is not upgradeable because FeatureGatesUpgradeable: "TechPreviewNoUpgrade" does not allow updates
	clusteroperator/operator-lifecycle-manager is not upgradeable because ClusterServiceVersions blocking minor version upgrades to 4.23 or major version upgrades to 5.0:
- maximum supported OCP version for openshift-logging/cluster-logging.v6.5.1 is 4.22
	clusteroperator/cloud-credential is missing
	clusteroperator/cluster-autoscaler is missing
	clusteroperator/console is missing
	clusteroperator/image-registry is missing
	clusteroperator/insights is missing
	clusteroperator/machine-api is missing
	clusteroperator/marketplace is missing
	clusteroperator/storage is missing


[must-gather      ] OUT 2026-08-04T13:24:03.358661369Z namespace/openshift-must-gather-vbqwh created
[must-gather      ] OUT 2026-08-04T13:24:03.65146491Z clusterrolebinding.rbac.authorization.k8s.io/must-gather-v8ksv created
[must-gather      ] OUT 2026-08-04T13:24:04.035749577Z pod for plug-in image registry.redhat.io/lvms4/lvms-must-gather-rhel9@sha256:92ba27b93c4bc0180b4b61cf3cfd40330755133b8abc45d888f2ee6b6f4b73b5 created
[must-gather      ] OUT 2026-08-04T13:24:04.035797046Z pod for plug-in image registry.redhat.io/openshift5/ptp-must-gather-rhel9@sha256:5a49c7d3587cac886a96a8a2f60413ab7963c633b039a314ba4433a6173b3706 created
[must-gather      ] OUT 2026-08-04T13:24:04.035967825Z pod for plug-in image registry.redhat.io/openshift-logging/cluster-logging-rhel9-operator@sha256:1b097b1a7630af0d78a2405d2cdd63282099c2e658a018f4b5080271d8c6f604 created
[must-gather      ] OUT 2026-08-04T13:24:04.158331141Z pod for plug-in image quay.io/openshift-release-dev/ocp-v5.0-art-dev@sha256:be149b15f5a5658e85f1eee8a715bcff9c66e5fa5866f5f9e58e09208a63a673 created
```

midu16 commented:
/verified by @midu16

midu16 commented:
/retest-required

ardaguclu commented:
/retest
